app/models/admin.rb

```diff
@@ -40,4 +43,43 @@ def send_set_password_notification
     token = set_reset_password_token
     AdminMailer.set_password_notification(self, token).deliver
   end
+
+  def to_param
+    login
+  end
+
+  serialize :otp_backup_codes, type: Array, coder: YAML, yaml: { permitted_classes: [String] }
+
+  attr_accessor :otp_plain_backup_codes
+
+  # Generate an OTP secret it it does not already exist
+  def generate_otp_secret_if_missing!
+    return unless otp_secret.nil?
+
+    update!(otp_secret: Admin.generate_otp_secret)
+  end
+
+  # Ensure that the user is prompted for their OTP when they login
+  def enable_otp!
+    update!(otp_required_for_login: true)
+  end
```

Comment:
I know this is from the tutorial but it's a bit strange to have this method but then everywhere just reads directly from `otp_required_for_login` to check whether it's enabled. TODO Bilka do I want to do anything about this though